modules/gdscript/tests/scripts/parser/warnings/unassigned_variable.gd

```diff
@@ -1,2 +1,4 @@
 func test():
-	var __
+	var unassigned
+	print(unassigned)
+	unassigned = "something" # Assigned only after use.
```

Comment:
Could you please expand the test? Something like:

```gdscript
var a
print(a) # Unassigned, warn.
if a: # Still unassigned, warn.
    a = 1
    print(a) # Assigned (dead code), don't warn.
print(a) # "Maybe" assigned, don't warn.
```

Reply:
Added this snippet to the test.